src/frontend/src/lib/components/hero/Hero.svelte

```diff
@@ -18,26 +19,68 @@
 	// We only want to display the "Sign-in" call to action on pages that actually are displaying any content in the Hero pane.
 	let heroContent = true;
 	$: heroContent = usdTotal || summary;
+
+	let isCollapsed = false;
+
+	const setCollapse = (collapse: boolean) => {
+		if (collapseDisabled) {
+			return;
+		}
+		isCollapsed = $authSignedIn && collapse;
+	};
+
+	let collapseDisabled = false;
+	const enableScrollHandler = () => (collapseDisabled = false);
+	const disableScrollHandler = () => (collapseDisabled = true);
+
+	let y = 0;
+	let lastY = 0;
+
+	const handleChangeY = () => {
+		const dy = lastY - y;
+		lastY = y;
+		setCollapse(dy < 0);
+	};
+
+	$: y, handleChangeY();
+
+	// This part is necessary for non-touch devices when the scrolling is disabled once the header is
+	// collapsed (the content would fit the entire window without need of a scrollbar). In such cases,
+	// variable 'y' would not be refreshed since the user would try to scroll but can not.
+	const handleScroll = ({ deltaY }: WheelEvent) => setCollapse(deltaY > 0);
 </script>
 
-<div class={`hero pb-4 md:pb-6 ${background}`}>
-	{#if $pseudoNetworkChainFusion}
-		<ThreeBackground />
+<svelte:window bind:scrollY={y} on:wheel={handleScroll} />
```

Comment:
I did not consider that option! I went for the easiest path eheh! Thanks for the suggestions, changing it

Reply:
Great.

For inspiration, checkout the similar implementation in https://github.com/dfinity/voluntary-recycling-credits/

Reply:
> Great.
> 
> If you were looking for inspiration regarding the intersection observer, checkout the similar implementation in https://github.com/dfinity/voluntary-recycling-credits/

Strange I cannot access that repo.

So, @peterpeterparker , I created a new `CollapseObserver` component, please have a look: I had to put it in `+layout` (outside the `Hero`) because the `observer` was not triggered back when the content would suddenly fit the entire page once collapsed. Did I explain correctly?

Reply:
Instead of using a component that injects a DOM element with no purpose, you can implement a Svelte directive and attach it to an existing element. This approach is a bit cleaner.

I've added you to https://github.com/dfinity/voluntary-recycling-credits, so you should now have access.

There's also another example of a directive using the Intersection Observer in https://github.com/dfinity/nns-dapp

WDYT?

Reply:
So, I tried a few things based on that repo, however:

1. Since we don't customize events, I would need to put `use:` in an existing `<div>` element somewhere and then execute the dispatched event on the parent. 
2. However, it cannot be inside `Hero` or any sub-component of `Hero`, because it will not re-trigger the de-collapse, in the case where the content fits the entire page (same reason as before), see video below.
3. Therefore, I think it must be in the `+layout` page, as a separate component.
4. Do you have a better idea/solution?

https://github.com/user-attachments/assets/b82d5a94-7e05-4bae-bf59-0f64ca2c4b4b

Reply:
1. Not sure what you mean with "the dispatched event on the parent". If I remember correctly, the two solutions I shared are tight with global stores so not sure where the event needs to bubble.

2. I'm not sure what's "in the case where the content fits the entire page". Which content?

Reply:
Let me give it a try based on your branch, I'll undestand better.

Reply:
Try like this: make it the list of tokens so that, once the header is collapsed, they all fit in the screen (there is no scrollbar anymore, nor the user can scroll)

Reply:
> 1. Not sure what you mean with "the dispatched event on the parent". If I remember correctly, the two solutions I shared are tight with global stores so not sure where the event needs to bubble.

Ah, I removed the store usage from the solutions you showed me: I thought we could use something simpler, since it is used just in this specific case. The solutions you gave me were using the intersections in a few other places, so it would make sense

Reply:
Actually the solution of vcr repo does not use a store but a global event. I'll try that.

Reply:
It mixes both: it uses a global event to set a store after. I just removed the store part, and make it so it directly set a variable in the component

Reply:
I gave it a try and now I understand what you mean with "in the case where the content fits the entire page" and also finally understand the last video your shared. The challenge is not to collapse the hero, regardless how to do so and what is the source of the trigger - a pseudo element or the title -, but it's to make the hero appear again.

I gave it a try by switching observers but, don't like the solution as there is a glitch. I created a PR that way you can have a look: https://github.com/dfinity/oisy-wallet/pull/2036

But I guess it's a dead end and unfortunatelly we have to switch back to scroll?

I'm sure I'm missing something but, cannot figure out what.

Now I better understand. I let you pave the route and decide what solution works best.

Reply:
Spontaneous idea to valide: we should change the layout and make the header absolute and the content `padding-top`.

Reply:
Actually, the more I think at it the more I think this is the correct way - i.e. the layout should be reviewed to integrate a collapsible header as the "hero" is not a "hero" anymore but, a header.